compiler/ast/src/expressions/literal.rs

```diff
@@ -32,13 +32,31 @@ pub enum Literal {
     /// A group literal, either product or affine.
     /// For example, `42group` or `(12, 52)group`.
     Group(Box<GroupLiteral>),
-    /// An integer literal, e.g., `42`.
-    Integer(IntegerType, String, #[serde(with = "leo_span::span_json")] Span),
     /// A scalar literal, e.g. `1scalar`.
     /// An unsigned number followed by the keyword `scalar`.
     Scalar(String, #[serde(with = "leo_span::span_json")] Span),
     /// A string literal, e.g., `"foobar"`.
     String(String, #[serde(with = "leo_span::span_json")] Span),
+    /// An 8-bit signed integer literal, e.g., `42i8`.
+    I8(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 16-bit signed integer literal, e.g., `42i16`.
+    I16(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 32-bit signed integer literal, e.g., `42i32`.
+    I32(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 64-bit signed integer literal, e.g., `42i64`.
+    I64(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 128-bit signed integer literal, e.g., `42i128`.
+    I128(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 8-bit unsigned integer literal, e.g., `42u8`.
+    U8(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 16-bit unsigned integer literal, e.g., `42u16`.
+    U16(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 32-bit unsigned integer literal, e.g., `42u32`.
+    U32(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 64-bit unsigned integer literal, e.g., `42u64`.
+    U64(String, #[serde(with = "leo_span::span_json")] Span),
+    /// A 128-bit unsigned integer literal, e.g., `42u128`.
+    U128(String, #[serde(with = "leo_span::span_json")] Span),
```

Comment:
Before, we defined the integer literals/types in a separate file.
I feel that the nesting the enums makes it more complicated to write code that processes integer literals and types.
I also feel that keeping everything in one file is simpler to look at.
Since it was a matter of personal preference, I don't mind switching back if it is preferred by the rest of team.